collection.go

```diff
@@ -259,19 +259,22 @@ func (cs *CollectionSpec) Assign(to interface{}) error {
 //
 // Returns an error if any of the fields can't be found, or
 // if the same Map or Program is assigned multiple times.
-func (cs *CollectionSpec) LoadAndAssign(to interface{}, opts *CollectionOptions) error {
+func (cs *CollectionSpec) LoadAndAssign(to interface{}, opts *CollectionOptions) (err error) {
 	loader, err := newCollectionLoader(cs, opts)
 	if err != nil {
 		return err
 	}
-	defer loader.cleanup()
 
-	// Support assigning Programs and Maps, lazy-loading the required objects.
 	assignedMaps := make(map[string]bool)
+	assignedProgs := make(map[string]bool)
+	defer loader.close(assignedMaps, assignedProgs, err)
```

Comment:
I don't know if this intentional, but this will fail to capture any changes to `err` after this point. See https://go.dev/play/p/BzoyZysNcyT

Reply:
Good catch, we need to close over `err` here to make sure it captures the return value. Will have to add a test for this.